daemon/src/model/cfd.rs

```diff
@@ -549,6 +579,43 @@ impl Cfd {
         Ok(signed_refund_tx)
     }
 
+    pub fn commit_tx(&self) -> Result<Transaction> {
+        let dlc = if let CfdState::Open { dlc, .. } | CfdState::PendingOpen { dlc, .. } =
```

Comment:
Could have another accessor or that returns `Option<Dlc>`, then you could just use `context` to bail out of the case where it is `None`.